### ISSUE TITLE
Sync Issue, CronJob Stuck at Running

### ISSUE THREAD

**Issue by alloylab** [mailchimp/mc-magento2#190]
## Issue Description

My store is showing up under connected sites in MailChimp site, however no data will sync. 
The ebizmarts_ecommerce job in the cron_schedule table shows it is still running after two days... I am trying this in a test system with only 3products, 4 customers, and 3 subscribers. The following tables also have no data in them for some reason, 
mailchimp_sync_batches
mailchimp_sync_ecommerce
mailchimp_webhook_request

### Preconditions
    1. Magento CE 2.2.1
    2. mc-magento2 1.0.26
    3. Setup MailChimp Module

### Steps to reproduce

Install modules via composer, 
"ebizmarts/mailchimp-lib": "~3"
"mailchimp/mc-magento2": "1.0.26"

In an existing environment, setup the module with the API key and once saved the store is displayed in mailchimp but no data is available. There have been no exceptions or php errors in the logs.. however MailChimp.log only has the following data in it,

[2018-01-29 13:45:45] MailChimpLogger.INFO: NULL
 [] []
[2018-01-29 13:46:22] MailChimpLogger.INFO: NULL
 [] []
[2018-01-29 14:00:03] MailChimpLogger.INFO: NULL
 [] []

**Comment by gonzaloebiz** [comment]
Hi @alloylab 

be sure that you have the cron configured.
Check the cron_schedule table

Best

**Comment by alloylab** [comment]
@gonzaloebiz cron is configured... from my original post, "The ebizmarts_ecommerce job in the cron_schedule table shows it is still running after two days" 

I am not sure I understand your statement about checking the cron_schedule table.

Thanks.

**Comment by gonzaloebiz** [comment]
Take a look at cron_schedule table for job_code = ebizmarts_ecommerce

Best

**Comment by alloylab** [comment]
here you go... the job is still running... with no content in mailchimp

![image](https://user-images.githubusercontent.com/1416037/35534386-fe243f40-0505-11e8-808e-21d8ddfa1f2c.png)

**Comment by gonzaloebiz** [comment]
Look at the php error log in the root directory of the user that run the magento cron system.

Best

**Comment by alloylab** [comment]
As I said originally "There have been no exceptions or php errors in the logs"... I guess I will just run xdebug and figure it out myself

**Comment by PiroozMB** [comment]
Hi. Did you ever solve this? I have exatly the same issue with the same version you mentioned.

**Comment by PiroozMB** [comment]
I'm having this issue with the version 1.0.27 too.

**Comment by gonzaloebiz** [comment]
Take a look at cron_schedule table for job_code = ebizmarts_ecommerce
If the job appear as "running", that generally means that the job has cancelled.
As the cron run as a php cli script, you must search for the php log (typically is in the document root directory of the use who run the cron)

Best

**Comment by jeffreytu** [comment]
I had same issue with the cron job stuck at running for ebizmarts_ecommerce. I followed the advice of @gonzaloebiz in the previous post and looked at my cron logs. First, my cron logs were not being output at all, so I made sure that I had my paths correct.

http://devdocs.magento.com/guides/v2.0/config-guide/cli/config-cli-subcommands-cron.html
```
* * * * * <path to php binary> <magento install dir>/bin/magento cron:run | grep -v "Ran jobs by schedule" >> <magento install dir>/var/log/magento.cron.log
* * * * * <path to php binary> <magento install dir>/update/cron.php >> <magento install dir>/var/log/update.cron.log
* * * * * <path to php binary> <magento install dir>/bin/magento setup:cron:run >> <magento install dir>/var/log/setup.cron.log
```
I specifically looked at the errors in update.cron. My specific issues were related to file permissions and php memory limits. Once I fixed those, truncated the cron_schedule table, and did cron:run, my MailChimp was syncing and updating. Now I'm fully synced.

Magento CE 2.2.1
MailChimp 1.0.27
